web/frontend/src/components/slam.vue

```diff
@@ -1,193 +1,208 @@
- *     rcLogConditionally(req);
- *     const chunks: Uint8Array[] = [];
- *     const getPointCloudMapStream = props.client.slamService.getPointCloudMapStream(req);
- *     getPointCloudMapStream.on('data', (res) => {
- *       const chunk = res.getPointCloudPcdChunk_asU8();
- *       chunks.push(chunk);
- *     });
- *     getPointCloudMapStream.on('status', (status) => {
- *       if (status.code !== 0) {
- *         const error = {
- *           message: status.details,
- *           code: status.code,
- *           metadata: status.metadata,
- *         };
- *         displayError(error);
- *       }
- *     });
- *     getPointCloudMapStream.on('end', (end) => {
- *       if (end === undefined || end.code !== 0) {
- *         // the error will be logged in the 'status' callback
- *         return;
- *       }
- *       pointcloud = concatArrayU8(chunks);
- *       pointCloudUpdateCount += 1;
- *     });
- *   });
- * };
- */
+const fetchSLAMMap = (name: string): Promise<Uint8Array> => {
+  return new Promise((resolve, reject) => {
+    const req = new slamApi.GetPointCloudMapStreamRequest();
+    req.setName(name);
+    rcLogConditionally(req);
+    const chunks: Uint8Array[] = [];
 
-/*
- * const fetchSLAMPose = (name: string) => {
- *   return nextTick(() => {
- *     const req = new slamApi.GetPositionNewRequest();
- *     req.setName(name);
- *     props.client.slamService.getPositionNew(req, new grpc.Metadata(), (error, res): void => {
- *       if (error) {
- *         displayError(error);
- *         return;
- *       }
- *       pose = res!.getPose()!;
- *     });
- *   });
- * };
- */
+    const getPointCloudMapStream = props.client.slamService.getPointCloudMapStream(req);
+    getPointCloudMapStream.on('data', (res) => {
+      const chunk = res.getPointCloudPcdChunk_asU8();
+      chunks.push(chunk);
+    });
+    getPointCloudMapStream.on('status', (status) => {
+      if (status.code !== 0) {
+        const error = {
+          message: status.details,
+          code: status.code,
+          metadata: status.metadata,
+        };
+        reject(error);
+      }
+    });
+    getPointCloudMapStream.on('end', (end) => {
+      if (end === undefined || end.code !== 0) {
+        // the error will be logged in the 'status' callback
+        return;
+      }
+      const arr = concatArrayU8(chunks);
+      resolve(arr);
+    });
+  });
+};
+
+const fetchSLAMPose = (name: string): Promise<commonApi.Pose> => {
+  return new Promise((resolve, reject): void => {
+    const req = new slamApi.GetPositionNewRequest();
+    req.setName(name);
+    props.client.slamService.getPositionNew(req, new grpc.Metadata(), (error, res): void => {
+      if (error) {
+        reject(error);
+        return;
+      }
+      resolve(res!.getPose()!);
+    });
+  });
+};
 
 const refresh2d = async (name: string) => {
+  const map = await fetchSLAMMap(name);
+  const returnedPose = await fetchSLAMPose(name);
+  return Promise.all([map, returnedPose]);
```

Comment:
This won't doing anything since we're `await`ing each function above in sequence. If you want to parallelize these requests then:
```ts
const result = await Promise.all([fetchSLAMMap(name), fetchSLAMPose(name)]);
return result;